### PR TITLE
EOL warnings for V2/V3 runtimes

### PR DESCRIPTION
**Pull request by nturinski**
Fixes https://github.com/microsoft/vscode-azurefunctions/issues/3676

I was going back and forth on the UI a lot here. Feedback welcomed. @ejizba @MadhuraBharadwaj-MSFT 

image

image

### CODE REVIEW

File: src/commands/createFunctionApp/stacks/FunctionAppStackStep.ts [hunk 1]
```diff
@@ -54,13 +54,15 @@ export class FunctionAppStackStep extends AzureWizardPromptStep<IFunctionAppWiza
...
     private async getPicks(context: IFunctionAppWizardContext): Promise<IAzureQuickPickItem<FullFunctionAppStack | undefined>[]> {
         const picks: IAzureQuickPickItem<FullFunctionAppStack | undefined>[] = await getStackPicks(context);
+        const majorVersion = getMajorVersion(context.version);
+        const isEol = Number(majorVersion) === 2 || Number(majorVersion) === 3;
         if (picks.length === 0) {
-            const majorVersion = getMajorVersion(context.version);
             const noPicksMessage = context.stackFilter ?
                 localize('noStacksFoundWithFilter', '$(warning) No stacks found for "{0}" on Azure Functions v{1}', context.stackFilter, majorVersion) :
                 localize('noStacksFound', '$(warning) No stacks found for Azure Functions v{0}', majorVersion);
+            const upgradeMessage = localize('eolWarning', '$(warning) No stacks found for Azure Functions v{0} due to being EOL. Upgrade your runtime version to see available stacks.')
```
Comment (alexweininger) [inline src/commands/createFunctionApp/stacks/FunctionAppStackStep.ts:63]: Forgot to pass `majorVersion` into the localize call

**Review by ejizba — APPROVED** [review]
Added some suggestions, but nothing blocking. Thanks!

**Comment by MadhuraBharadwaj-MSFT** [comment]
@nturinski it looks like we're using the 'warning symbol' for versions that are soon approaching end of life (Python 3.7 in your screenshot above) and the acronym 'EOL' for versions that have already approached end of life. Is this differentiation intended?
My suggestion is to use the 'warning symbol' across all the end-of-life notices irrespective of language versions/host versions that are already retired/soon approaching retirement state. I think this will help maintain consistency and serve as a visual reminder for users that they're using something that's either unsupported or soon-to-be unsupported.

I agree with other suggestions @ejizba made above.

Thank you for working on this @nturinski !

File: src/FuncVersion.ts [hunk 2]
```diff
@@ -16,14 +16,21 @@ export enum FuncVersion {
...
     const recommended: string = localize('recommended', '(Recommended)');
     let picks: IAzureQuickPickItem<FuncVersion | undefined>[] = [
         { label: 'Azure Functions v4', description: recommended, data: FuncVersion.v4 },
-        { label: 'Azure Functions v3', data: FuncVersion.v3 },
-        { label: 'Azure Functions v2', data: FuncVersion.v2 },
-        { label: 'Azure Functions v1', data: FuncVersion.v1 }
+        { label: 'Azure Functions v3', data: FuncVersion.v3, description: '(EOL)' },
+        { label: 'Azure Functions v2', data: FuncVersion.v2, description: '(EOL)' },
+        { label: 'Azure Functions v1', data: FuncVersion.v1 }, {
+            label: localize('runtimeWarning', `$(extensions-warning-message) Azure Functions runtime v2.x and v3.x have reached the EOL. Learn more...`),
```
Comment (bwateratmsft) [inline src/FuncVersion.ts:28]: The indendation here was confusing at first. Maybe put the opening brace on line 27 on its own line?

Comment (nturinski) [reply src/FuncVersion.ts:28]: Ugh yeah, I really hate how it creates another new line for the object, but I can see why that's confusing

File: src/commands/createFunctionApp/stacks/FunctionAppStackStep.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (nturinski) [reply src/commands/createFunctionApp/stacks/FunctionAppStackStep.ts:63]: That makes sense. I guess knowing the migration guide isn't that useful when you're selecting a runtime version for the first time.

**Comment by nturinski** [comment]
New UI changes.

image
image

**Review by ejizba — APPROVED** [review]

**Pull request merged**